eZ/Bundle/EzPublishCoreBundle/Command/ReindexCommand.php

```diff
@@ -54,7 +60,11 @@ class ReindexCommand extends ContainerAwareCommand
     public function initialize(InputInterface $input, OutputInterface $output)
     {
         parent::initialize($input, $output);
-        $this->searchIndexer = $this->getContainer()->get('ezpublish.spi.search.indexer');
+
+        $searchIndexerFactory = $this->getContainer()->get('ezpublish.api.search_engine.indexer.factory');
+        $this->errorHandler = new LocalErrorHandler($input->getOption('continue-on-error'));
+        $this->searchIndexer = $searchIndexerFactory->buildSearchEngineIndexerWithErrorHandler($this->errorHandler);
```

Comment:
Would perhaps be clearer if either:
A. get indexer like before and simply call the new setter method for setting the optional error handler
B. Maybe remove the continue-on-error handling from ErrorHandler so it can be injected by DI? If so it will remove the hard coding of handler here and will instead need continue-on-error flag to be sent to indexer
C. ..

wdyt @alongosz & @adamwojs ?

Reply:
Search Indexer with proper error handler (collector? :grinning: ) should be injected via DI.

Reply:
On second thought, since building this service depends on `continue-on-error` command option, it might be impossible to do it completely via DI. However still factories shouldn't be used directly in a such way.
So I'd rather simplify this to:
```php
$this->searchIndexer->setErrorHandler(new LocalErrorHandler($input->getOption('continue-on-error')))
````
Of course instead of `new LocalErrorHandler` you could create factory for error handler, but that's maybe an overkill, since only `LocalErrorHandler` provides strategy to continue on errors.

To be safe, before calling setErrorHandler I'd heck if `$searchIndexer instanceof IncrementalIndexer`.

Side note: could you refactor this command to get only needed services instead of entire container? When I wrote it I lacked proper knowledge :sweat_smile:

Reply:
This is A.

B would be to move continue-on-error logic injection to indexer or somewhere else so collector does not need to know about this, and instead just collects if called.